mr/src/main/java/org/apache/iceberg/mr/hive/HiveOutputFileFactory.java

```diff
@@ -0,0 +1,50 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.iceberg.mr.hive;
+
+import org.apache.hadoop.mapred.TaskAttemptID;
+import org.apache.iceberg.FileFormat;
+import org.apache.iceberg.PartitionSpec;
+import org.apache.iceberg.encryption.EncryptionManager;
+import org.apache.iceberg.io.FileIO;
+import org.apache.iceberg.io.LocationProvider;
+import org.apache.iceberg.io.OutputFileFactory;
+
+public class HiveOutputFileFactory extends OutputFileFactory {
+  private final TaskAttemptID taskAttemptID;
+
+  public HiveOutputFileFactory(PartitionSpec spec, FileFormat fileFormat, LocationProvider locationProvider, FileIO io,
+                               EncryptionManager encryptionManager, TaskAttemptID taskAttemptID) {
+    super(spec, fileFormat, locationProvider, io, encryptionManager, 0, 0);
+    this.taskAttemptID = taskAttemptID;
+  }
+
+  /**
+   * Override the filename generation so it contains jobId, taskId, taskAttemptId. Kept the UUID and the fileCount so
+   * the filenames are similar for other writers.
+   * @return The generated file name
+   */
+  @Override
+  protected String generateFilename() {
+    return format().addExtension(
+        String.format("%05d-%d-%d-%s-%05d", taskAttemptID.getJobID().getId(), taskAttemptID.getTaskID().getId(),
```

Comment:
The first number is typically the task ID so we can trace back to the task that produced a file easily.

The job ID is constant and will be 1-to-1 with a write UUID, so I'm not sure that it is needed. If not, then this could use the existing factory and pass `taskAttemptId.getTaskID()` and `taskAttemptId.getId()` to the constructor (instead of 0, 0).

I think I'd prefer using the existing one and not creating a new subclass here.

Reply:
The uuid is generated by the OutputFileFactory - this will be generated on every mapper-reducer. So different tasks / even task attempts will have a different uuid. The only thing connecting together the files generated by the same query would be the jobID (and even that can restart).
I was thinking about changing the fileName format to something like:
`<queryId>-<jobId>-<taskId>-<attemptId>-<fileCount>.<fileFormat>`
But decided to keep something more like the current names for the familiarities shake.

Reply:
The UUID was supposed to be a UUID for the write, not for each task. Looks like that was broken at some point by refactoring.

In that case, What do you think about adding a constructor that accepts a UUID string? Then you could call it with an identifier that works for your job, whether that's a job-level UUID from config or a Hadoop Job ID. And we can follow up in the other engines and generate the UUID correctly.

Reply:
Removed HiveOutputFileFactory - added new cosntructor to OutputFileFactory instead

Also since the locations are generated by the OutputFileFactory, we do not need LocationHelper anymore, so I removed it as well.